File: Mult_Digits/Mult.cpp
/*
    Author : Pakshal Jain
    Date : 20/10/2022
    Program : To Calculate Multiplication of Digits of Entered Number
*/

//Required Libraries
#include<iostream>
using namespace std;

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class Multi
{
    //Characteristics of Class
    private :
        int No = 0;
    //Constructor of Class
    public :
        Multi(int no)
        {
            this->No = no;
        }
    //Behaviour of Class
    int Calculate()
    {
        int Mult = 1;
        int Digit = 0;

        //Logic To Calculate Multiplication of Digits
        while(this->No != 0)
        {
            Digit = No % 10;
            if(Digit == 0)
            {
                Digit = 1;
            }
            Mult = Mult * Digit;
            No = No / 10;
        }

        return Mult;
    }
};

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//Main Function

int main() 
{
    int No = 0;
    int iret = 0;

    cout<<"Jay Ganesh...\n";
    
    //Taking Input From User
    cout<<"Enter One Number\n";
    cin>>No;

    //Crating object of Multi Class
    Multi mobj(No);
    //Call To Behaviour of Class
    iret = mobj.Calculate();

    //Printing Multiplication of Digits
    cout<<"Multiplication of Digits of Entered Number is : "<<iret;

    return 0;
}

File: Mult_Digits/Mult.go
/*
	Auhtor : Pakshal Jain
	Date : 20/10/2022
	Program : To Calculate Multiplication of Digits of Entered Number
*/

//Required Libraries
package main
import "fmt"

//Function To Calculate Multiplication of Digits
func Multiplication(No int) int {
	var Digit int = 0;
	var Mult int = 1;

	//Logic To Calculate Multiplication of Digits
	for No != 0 {
		Digit = No % 10;
		if Digit == 0 {
			Digit = 1;
		}
		Mult = Mult * Digit;
		No = No / 10;
	}

	return Mult;
}

//Main Function
func main() {
	var No int = 0;
	var iret int = 0;

	fmt.Println("Jay Ganesh...");

	fmt.Println("Enter One Number");
	fmt.Scanf("%d",&No);

	//Call To Main Function
	iret = Multiplication(No);

	//Printing Multiplication of Digits
	fmt.Println("Multiplication of Digits of Entered Number is : ",iret);
}